assets/src/edit-story/components/panels/shared/flipControls.js

```diff
@@ -18,38 +18,82 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
+import { useCallback, useEffect, useState } from 'react';
 
 /**
  * Internal dependencies
  */
 import { ReactComponent as FlipHorizontal } from '../../../icons/flip_horizontal.svg';
 import { ReactComponent as FlipVertical } from '../../../icons/flip_vertical.svg';
 import Toggle from '../../form/toggle';
+import getCommonObjectValue from '../utils/getCommonObjectValue';
+import { getDefinitionForType } from '../../../elements';
+
+function FlipControls({ selectedElements, onSetProperties }) {
+  const flip = getCommonObjectValue(
+    selectedElements,
+    'flip',
+    ['horizontal', 'vertical'],
+    false
+  );
+  const [state, setState] = useState({
+    flip,
+  });
+  useEffect(() => {
+    setState({ flip });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [flip.horizontal, flip.vertical]);
+
+  const canFlip = selectedElements.every(
+    ({ type }) => getDefinitionForType(type).canFlip
+  );
+
+  const updateProperties = useCallback(() => {
+    onSetProperties(({ flip: oldFlip, type }) => {
+      const update = {
+        ...state,
+        flip:
+          // Ensure flip change only if flip controls are actually visible (canFlip).
+          canFlip && getDefinitionForType(type).canFlip ? state.flip : oldFlip,
```

Comment:
`&& getDefinitionForType(type).canFlip` is this check necessary? Wouldn't that always be covered by `canFlip`?

Reply:
Indeed! Changed in a8cd9760